src/gen_apple/ga_dev_gps.cpp

```diff
@@ -23,17 +27,21 @@ void ga_dev_gps_open(void) {
 
 }
 
-void ga_dev_gps_high(void){
-    if(digitalRead(_PIN_GPS_ENABLE) == HIGH){
-      return;
+/* Function to turn GPS on or off */
+void ga_dev_gps_power(void){
+    if(digitalRead(_GA_PIN_GPS_ENABLE)== HIGH){
```

Comment:
Running digitalRead() on a pin that is configured as a output pin will not give back the state that the pin is in. You'll need to keep track of the state yourself manually. 

Here is a small segment that I tried, just to make sure:

```
    pinMode(9, INPUT);

    digitalWrite(9, HIGH);
    state = digitalRead(9);
    Serial.println(state);

    digitalWrite(9, LOW);
    state = digitalRead(9);
    Serial.println(state);

```

Here is the output:

```
1
1
```

Also, I recommend you change the function so that you can pass the state that you want the gps_power to be: that way it's easier to read through the code an understand what is happening. In the case where you have ga_dev_gps_power in multiple places, how would you know what state the device is actually in?


```
#define ON 1
#define OFF 0

void ga_dev_gps_power(int state){
    if( state == ON){
    ...
    }
    else {
    ...
    } 
}

ga_dev_gps_power(ON);

```

If you want to be fancy you can also use enum to do this:

```
enum PowerState {
  ON,
  OFF
}

void ga_dev_gps_power(PowerState state){
    if( state == _ON){
    ...
    }
    else {
    ...
    } 
}

ga_dev_gps_power(_ON);


```